vernac/lemmas.ml

```diff
@@ -193,16 +193,38 @@ let start_lemma_with_initialization ?hook ~poly ~scope ~kind ~udecl sigma recgua
         | None -> p
         | Some tac -> pi1 @@ Proof.run_tactic Global.(env ()) tac p)) lemma
 
-let start_lemma_com ~program_mode ~poly ~scope ~kind ?inference_hook ?hook thms =
+let obligations_default_tac = ref (Proofview.tclUNIT)
+
+let inference_hook ~program_mode =
+  if program_mode then
```

Comment:
why not remove the if here and push the logic to `start_lemma_com`?

Reply:
why change it?

Reply:
It seems strange to have a conditional hook, I'd rather don't set the hook at all if not needed.